> `openvizsla_export.h` definitely must not be a part of the public interface. What issue do you try to solve? Everything seems to build fine with VS 2015 and 2019 for me.

CMake documentation seems to disagree: "The resulting file should be installed with other headers in the library." [1]

The problem this fixes is:
```
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(110,19): error C2375: 'ov_new': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(53): message : see declaration of 'ov_new' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(153,5): error C2375: 'ov_open': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(54): message : see declaration of 'ov_open' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(204,6): error C2375: 'ov_free': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(55): message : see declaration of 'ov_free' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(212,5): error C2375: 'ov_get_usb_speed': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(57): message : see declaration of 'ov_get_usb_speed' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(225,5): error C2375: 'ov_set_usb_speed': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(58): message : see declaration of 'ov_set_usb_speed' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(238,5): error C2375: 'ov_capture_start': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(60): message : see declaration of 'ov_capture_start' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(262,5): error C2375: 'ov_capture_dispatch': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(61): message : see declaration of 'ov_capture_dispatch' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(273,5): error C2375: 'ov_capture_stop': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(62): message : see declaration of 'ov_capture_stop' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(283,5): error C2375: 'ov_load_firmware': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(64): message : see declaration of 'ov_load_firmware' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\src\ov.c(325,13): error C2375: 'ov_get_error_string': redefinition; different linkage [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
       C:\Development\vcpkg\buildtrees\libopenvizsla\src\0.1.0-rc2-849b5d1270\include\ov.h(66): message : see declaration of 'ov_get_error_string' [C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj]
     5>Done Building Project "C:\Development\vcpkg\buildtrees\libopenvizsla\x86-windows-dbg\openvizsla.vcxproj" (default targets) -- FAILED.
```

[1] https://cmake.org/cmake/help/v3.7/module/GenerateExportHeader.html

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -82,7 +82,7 @@ install:
 before_build:
   - mkdir build
   - cd build
-  - cmake -G"%GENERATOR%" -DCMAKE_TOOLCHAIN_FILE=c:/tools/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=%VCPKG_TARGET_TRIPLET% -DCMAKE_BUILD_TYPE=%CONFIGURATION% ..
+  - cmake -G"%GENERATOR%" -DCMAKE_TOOLCHAIN_FILE=c:/tools/vcpkg/scripts/buildsystems/vcpkg.cmake -DVCPKG_TARGET_TRIPLET=%VCPKG_TARGET_TRIPLET% -DCMAKE_BUILD_TYPE=%CONFIGURATION% -DBUILD_SHARED_LIBS:BOOL=on ..
 build_script:
   - cmake --build . --config %CONFIGURATION%
 test_script:

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,6 +131,8 @@ install(TARGETS openvizsla
 	DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
 install(FILES "include/ov.h"
 	DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/openvizsla)
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/openvizsla_export.h"
+	DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/openvizsla)
 configure_file("openvizsla.pc.in" "openvizsla.pc" @ONLY)
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/openvizsla.pc"
 	DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig")

--- a/include/ov.h
+++ b/include/ov.h
@@ -6,6 +6,8 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <openvizsla_export.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
@@ -50,20 +52,20 @@ enum ov_usb_speed {
 	OV_HIGH_SPEED = 0x48
 };
 
-struct ov_device* ov_new(const char* firmware_filename);
-int  ov_open(struct ov_device* ov);
-void ov_free(struct ov_device* ov);
+OPENVIZSLA_EXPORT struct ov_device* ov_new(const char* firmware_filename);
+OPENVIZSLA_EXPORT int  ov_open(struct ov_device* ov);
+OPENVIZSLA_EXPORT void ov_free(struct ov_device* ov);
 
-int ov_get_usb_speed(struct ov_device* ov, enum ov_usb_speed* speed);
-int ov_set_usb_speed(struct ov_device* ov, enum ov_usb_speed speed);
+OPENVIZSLA_EXPORT int ov_get_usb_speed(struct ov_device* ov, enum ov_usb_speed* speed);
+OPENVIZSLA_EXPORT int ov_set_usb_speed(struct ov_device* ov, enum ov_usb_speed speed);
 
-int ov_capture_start(struct ov_device* ov, struct ov_packet* packet, size_t packet_size, ov_packet_decoder_callback callback, void* user_data);
-int ov_capture_dispatch(struct ov_device* ov, int count);
-int ov_capture_stop(struct ov_device* ov);
+OPENVIZSLA_EXPORT int ov_capture_start(struct ov_device* ov, struct ov_packet* packet, size_t packet_size, ov_packet_decoder_callback callback, void* user_data);
+OPENVIZSLA_EXPORT int ov_capture_dispatch(struct ov_device* ov, int count);
+OPENVIZSLA_EXPORT int ov_capture_stop(struct ov_device* ov);
 
-int ov_load_firmware(struct ov_device* ov, const char* filename);
+OPENVIZSLA_EXPORT int ov_load_firmware(struct ov_device* ov, const char* filename);
 
-const char* ov_get_error_string(struct ov_device* ov);
+OPENVIZSLA_EXPORT const char* ov_get_error_string(struct ov_device* ov);
 
 #ifdef __cplusplus
 }

